KEDamage.cs

```diff
@@ -8,58 +8,73 @@ class KEDamage:Damage
     {
         public struct KineticData
         {
+            /// <summary>
+            /// The pierce of the shot
+            /// </summary>
             public float Pierce;
-
+            /// <summary>
+            /// Distance between the units
+            /// </summary>
             public float Distance;
-
-            public float ArmorKEFactor;
-
-            public float HealthKEFactor;
-
+            /// <summary>
+            /// Multiplier for armor degradation
+            /// </summary>
+            public float ArmorDegradationFactor;
+            /// <summary>
+            /// Multiplier for health damage
+            /// </summary>
+            public float HealthDamageFactor;
+            /// <summary>
+            /// Air friction constant used in calculation of attenuation
+            /// </summary>
             public float FrictionFactor;
         }
 
-        private KineticData data;
-        private Damage.Target target;
+        private KineticData _keData;
 
-        public KEDamage(KineticData d,Damage.Target t):base(EDamageTypes.KE)
+        public KEDamage(KineticData d,Damage.Target t):base(EDamageTypes.KE, t)
         {
-            data = d;
-            target = t;
+            _keData = d;
         }
 
-        override public float DealDamage()
+        public override Target CalculateDamage()
         {
+            Target finalState = _target;
```

Comment:
Shot in the dark (because I haven't studied your code all the way)--this just creates a copy of _target. You might want to just manipulate _target than create another instance (copy).